archinfo/arch_arm.py

```diff
@@ -336,11 +336,32 @@ class ArchARMCortexM(ArchARMEL):
         # TODO: POP.W
     }
 
-    def __init__(self, *args, **kwargs):
+    # Special handling of CM mode in *stone
+    if _capstone:
+        cs_arch = _capstone.CS_ARCH_ARM
+        cs_mode = _capstone.CS_MODE_LITTLE_ENDIAN + _capstone.CS_MODE_THUMB + _capstone.CS_MODE_MCLASS
+    _cs_thumb = None
+    if _keystone:
+        ks_arch = _keystone.KS_ARCH_ARM
+        ks_mode = _keystone.KS_MODE_THUMB + _keystone.KS_MODE_LITTLE_ENDIAN
+    _ks_thumb = None
+
+    @property
+    def capstone_thumb(self):
+        return self.capstone
+
+    @property
+    def keystone_thumb(self):
+        return self.keystone
 
         # See http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0553a/CHDBIBGJ.html
         # for relevant pain-points.  Most of these only get updated on an exception/interrupt/syscall
         # Except we leave T on because we have to.
+
+    # EDG says: Note, due to the thing about class members being single objects, don't do this
+    # in the constructor, or you're gonna have a bad time
```

Comment:
But you _are_ doing this in the constructor. Do you mean in the class body? You can just say `register_list = ArchARM.register_list + [...]`

Reply:
@rhelmot but you cactually can't -- that's a class object, so each time you make an ArchARMCortexM() you'll add those to the list, and get lots of duplicates. 

I solved the problem by duplicating the list.  I also simplified it and made it more realistic while I'm at it.